### PR TITLE
Buttons enhanced

### PR DESCRIPTION
**Pull request by bell07**
Added image_button (with tooltip) and item_image_button. Added the methods needed to use it.

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -26,6 +26,8 @@
 * state:onInput( function(self,fields,playername)) - specify a function to run after data and/or events received
 * state:button( x,y,w,h,name,text [, exit_on_click] ) - create a new button at x,y with name and caption (text)
  * ^ optional: exit_on_click - set to true to exit the form when the button is clicked. ( Also see button.setClose() )
+   state:image_button( x,y,w,h,name,text, image [, exit_on_click] ) create a new button with image.
```
Comment (kaeza) [inline docs/API.md]: Missing asterisk (`*`)? Same below.

As-is:

* state:button( x,y,w,h,name,text [, exit_on_click] ) - create a new button at x,y with name and caption (text)
￼   * ^ optional: exit_on_click - set to true to exit the form when the button is clicked. ( Also see button.setClose() )
￼   state:image_button( x,y,w,h,name,text, image [, exit_on_click] ) create a new button with image.
   state:item_image_button( x,y,w,h,name,text, item [, exit_on_click] ) create a new button with item as image.

With asterisks:

* state:button( x,y,w,h,name,text [, exit_on_click] ) - create a new button at x,y with name and caption (text)
￼   * ^ optional: exit_on_click - set to true to exit the form when the button is clicked. ( Also see button.setClose() )
* state:image_button( x,y,w,h,name,text, image [, exit_on_click] ) create a new button with image.
* state:item_image_button( x,y,w,h,name,text, item [, exit_on_click] ) create a new button with item as image.

**Comment by bell07** [comment]
It was a typo. Thanks for the hint. Fixed now.

**Comment by rubenwardy** [comment]
Looks good! Willing for this to be merged?

**Comment by bell07** [comment]
I am willing since 18 days ago ;-) The same for the second open PR. But I think you intend a feedback from any other "minetest-mods" team member :|

**Comment by rubenwardy** [comment]
I'm quite busy IRL, so tend to forget about these things

**Pull request merged**